worlds/banjo_tooie/__init__.py

```diff
@@ -0,0 +1,1094 @@
+            raise OptionError("You cannot have progressive Water Training\
+                without randomizing moves and randomizing BK moves")
+        if self.options.progressive_flight\
+                and (not self.options.randomize_bk_moves or not self.options.randomize_bt_moves):
+            raise OptionError("You cannot have progressive flight without randomizing moves and randomizing BK moves")
+        if self.options.progressive_egg_aiming != ProgressiveEggAim.option_none\
+                and (not self.options.randomize_bk_moves or not self.options.randomize_bt_moves):
+            raise OptionError(
+                "You cannot have progressive egg aiming without randomizing moves and randomizing BK moves"
+            )
+        if self.options.progressive_bash_attack\
+                and (not self.options.randomize_stop_n_swap or not self.options.randomize_bt_moves):
+            raise OptionError(
+                "You cannot have progressive bash attack without randomizing Stop N Swap and randomizing BK moves"
+                )
+        if not self.options.randomize_bt_moves and self.options.jamjars_silo_costs != JamjarsSiloCosts.option_vanilla:
+            raise OptionError("You cannot change the silo costs without randomizing Jamjars' moves.")
+        if not self.options.open_hag1\
+                and self.options.victory_condition == VictoryCondition.option_wonderwing_challenge:
+            self.options.open_hag1.value = True
+        if self.options.world_requirements != WorldRequirements.option_normal and not self.options.skip_puzzles:
+            raise OptionError("Your world requirements needs to be set to normal if you are not going to skip puzzles.")
+
+    def choose_starter_egg(self) -> None:
+        if self.options.egg_behaviour == EggsBehaviour.option_random_starting_egg or \
+                self.options.egg_behaviour == EggsBehaviour.option_simple_random_starting_egg:
+            eggs: list = []
+            if self.options.egg_behaviour == EggsBehaviour.option_random_starting_egg:
+                eggs = list([itemName.BEGGS, itemName.FEGGS, itemName.GEGGS, itemName.IEGGS, itemName.CEGGS])
+            else:
+                eggs = list([itemName.BEGGS, itemName.FEGGS, itemName.GEGGS, itemName.IEGGS])
+            egg_name = self.random.choice(eggs)
+            starting_egg = self.create_item(egg_name)
+            self.multiworld.push_precollected(starting_egg)
+            banjoItem = all_item_table.get(egg_name)
+            self.starting_egg = banjoItem.btid
+        else:
+            starting_egg = self.create_item(itemName.BEGGS)
+            self.multiworld.push_precollected(starting_egg)
+            banjoItem = all_item_table.get(itemName.BEGGS)
+            self.starting_egg = banjoItem.btid
+
+    def choose_starter_attack(self) -> None:
+        if self.options.randomize_bk_moves != RandomizeBKMoveList.option_none:
+            chosen_attack: str
+            base_attacks: list
```

Comment:
```suggestion
            chosen_attack: str
            base_attacks: list
```
These annotations aren't really necessary, and typing just as `list` without saying the contained type actually prevents it from being inferred in the assignments below.

Reply:
Annotations don't actually do a declaration in python, they're just to show types of values (unless it's something like dataclasses that look at the annotations and generate values from them). if/else blocks don't make their own scope, so they'd be visible after.

Reply:
Allright, removed both of them.